Podcast/Cache.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  Cache.swift
+//  Podcast
+//
+//  Created by Drew Dunne on 11/1/17.
+//  Copyright © 2017 Cornell App Development. All rights reserved.
+//
+
+import UIKit
+import SwiftyJSON
+
+class Cache: NSObject {
+    static let sharedInstance = Cache()
+    
+    // NOTE: We might want to thing about limiting cache size in the future,
+    // but not something to worry about now. 
+    private var episodeCache: [String: Episode]!
+    private var seriesCache: [String: Series]!
+    private var userCache: [String: User]!
+    
+    private override init() {
+        episodeCache = [:] // TODO: in future maybe store cache!
+        seriesCache = [:]
+        userCache = [:]
+    }
+    
+    func reset() {
+        // TODO: this probably shouldn't be called. Gotta test what happens to views.
+        // Possible error: the cache is reset, but views retain references to objects,
+        // thus causing errors when cache is refilled. Views not flushed will be out
+        // of sync with episodes loaded into reset cache from another view's endpoint
+        // call. 
+        episodeCache = [:]
```

Comment:
if there is an issue for this then let's take out resetting the episodeCache here just in case. So keep the function stub but have it be an empty call

Reply:
Alright can do.